hasura.planx.uk/migrations/1730742914548_create_index_bops_applications_session_id/up.sql

```diff
@@ -0,0 +1,14 @@
+CREATE  INDEX "bops_applications_session_id_idx" on
+  "public"."bops_applications" using hash ("session_id");
+CREATE  INDEX "uniform_applications_session_id_idx" on
+  "public"."uniform_applications" using hash ("submission_reference");
+CREATE  INDEX "email_applications_session_id_idx" on
+  "public"."email_applications" using hash ("session_id");
+CREATE  INDEX "s3_applications_session_id_idx" on
+  "public"."s3_applications" using hash ("session_id");
+CREATE  INDEX "feedback_flow_id_team_id_idx" on
+  "public"."feedback" using btree ("team_id", "flow_id");
+CREATE  INDEX "reconciliation_requests_session_id_idx" on
+  "public"."reconciliation_requests" using hash ("session_id");
+CREATE  INDEX "published_flows_created_at_idx" on
+  "public"."published_flows" using btree ("created_at");
```

Comment:
A couple notes about this migration:
- Why a mix of `hash` versus `btree` index types? 
  - I'm using `hash` indices when it's a column that is exclusively queried via an equality check and `btree` indices when a) it's a column that is queried via equality _or range_ checks or b) multiple columns are combined into the same index
  - Postgres docs on this are helpful https://www.postgresql.org/docs/current/indexes-types.html
- I'm adding an index to `published_flows.created_at` because that is how all of our current queries are ordering, but I have _not_ actually checked/compared yet if there's a difference to sort by published flow id (or jogged my memory about why we don't do this already)

Reply:
It may also be worth adding a `DESC` in here as we're always searching in this order ([docs](https://www.postgresql.org/docs/current/indexes-ordering.html)).

Reply:
Cool good shout - that's now updated :+1: Worth noting order isn't possible to specify in Hasura GUI for adding indexes, but worked no problem when added via "SQL" panel directly
![Screenshot from 2024-11-05 09-13-40](https://github.com/user-attachments/assets/ef7f68f7-9b71-45fb-8b62-8d3ea9e3508f)